packages/expo-device/ios/EXDevice/EXDevice.m

```diff
@@ -0,0 +1,252 @@
+typedef NS_ENUM(NSInteger, DeviceType) {
+  DeviceTypeHandset,
+  DeviceTypeTablet,
+  DeviceTypeTv,
+  DeviceTypeUnknown
+};
+
+#define DeviceTypeValues [NSArray arrayWithObjects: @"Handset", @"Tablet", @"Tv", @"Unknown", nil]
+
+- (DeviceType)getDeviceType
+{
+  switch ([[UIDevice currentDevice] userInterfaceIdiom]) {
+    case UIUserInterfaceIdiomPhone: return DeviceTypeHandset;
+    case UIUserInterfaceIdiomPad: return DeviceTypeTablet;
+    case UIUserInterfaceIdiomTV: return DeviceTypeTv;
+    default: return DeviceTypeUnknown;
+  }
+}
+
+- (bool) isTablet
+{
+  return [self getDeviceType] == DeviceTypeTablet;
+}
+
+- (NSString *)getCPUType {
+  /* https://stackoverflow.com/questions/19859388/how-can-i-get-the-ios-device-cpu-architecture-in-runtime */
+  const NXArchInfo *info = NXGetLocalArchInfo();
+  NSString *typeOfCpu = [NSString stringWithUTF8String:info->description];
+  return typeOfCpu;
+}
+
+
+- (unsigned long long) totalMemory {
+  return [NSProcessInfo processInfo].physicalMemory;
+}
+
+- (NSDictionary *) getStorageDictionary {
+  NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+  return [[NSFileManager defaultManager] attributesOfFileSystemForPath:[paths lastObject] error: nil];
+}
+
+- (uint64_t) totalDiskCapacity {
+  uint64_t totalSpace = 0;
+  NSDictionary *storage = [self getStorageDictionary];
+  
+  if (storage) {
+    NSNumber *fileSystemSizeInBytes = [storage objectForKey: NSFileSystemSize];
+    totalSpace = [fileSystemSizeInBytes unsignedLongLongValue];
+  }
+  return totalSpace;
+}
+
+- (uint64_t) freeDiskStorage {
+  uint64_t freeSpace = 0;
+  NSDictionary *storage = [self getStorageDictionary];
+  
+  if (storage) {
+    NSNumber *freeFileSystemSizeInBytes = [storage objectForKey: NSFileSystemFreeSize];
+    freeSpace = [freeFileSystemSizeInBytes unsignedLongLongValue];
+  }
+  return freeSpace;
+}
+
+
+- (NSDictionary *)constantsToExport
+{
+  UIDevice *currentDevice = [UIDevice currentDevice];
+  NSString *uniqueId = [DeviceUID uid]; //TODO: need to import this
+  
+  return @{
+           @"brand": @"Apple",
+           @"carrier": self.carrier ?: [NSNull null],
+           @"deviceType": [DeviceTypeValues objectAtIndex: [self getDeviceType]] ?: [NSNull null],
+           @"deviceName": currentDevice.name, //TODO, ADD TO JS AS WELL
+           @"deviceId": self.deviceId ?: [NSNull null],
+           //           @"freeDiskStorage": @(self.freeDiskStorage),
+           //           @"isEmulator": @NO,
+           @"isTablet": @(self.isTablet),
+           @"manufacturer": @"Apple",
+           //           @"model": self.deviceId, // DON'T WORRY ABOUT FOR NOW
+           //           @"phoneNumber": @"undefined3", // ANDROID ONLY
+           //           @"serialNumber": @"undefined4", // ANDROID ONLY
+           @"supportedABIs": @[[self getCPUType]],
+           @"systemName": currentDevice.systemName,
+           @"totalMemory": @(self.totalMemory),
```

Comment:
I think it would be clearer if, since `totalMemory` is a method, you'd use `[self totalMemory]` notation.